DAOS-15937 test: Automate metadata duplicate rpc detection time consuming

dinghwah opened the pull request:
Automate metadata duplicate rpc detection time consuming

Skip-unit-tests: true
Skip-fault-injection-test: true
Skip-func-hw-test-large-md-on-ssd: false
Test-tag: test_metadata_dup_rpc
Doc-only: false
Required-githooks: true

Signed-off-by: Ding Ho <ding-hwa.ho@intel.com>

### Before requesting gatekeeper:

* [ ] Two review approvals and any prior change requests have been resolved.
* [ ] Testing is complete and all tests passed or there is a reason documented in the PR why it should be force landed and forced-landing tag is set.
* [ ] `Features:` (or `Test-tag*`) commit pragma was used or there is a reason documented that there are no appropriate tags for this PR.
* [ ] Commit messages follows the guidelines outlined [here](https://daosio.atlassian.net/wiki/spaces/DC/pages/11133911069/Commit+Comments).
* [ ] Any tests skipped by the ticket being addressed have been run and passed in the PR.

### Gatekeeper:

* [ ] You are the appropriate gatekeeper to be landing the patch.
* [ ] The PR has 2 reviews by people familiar with the code, including appropriate owners.
* [ ] Githooks were used. If not, request that user install them and check copyright dates.
* [ ] Checkpatch issues are resolved.  Pay particular attention to ones that will show up on future PRs.
* [ ] All builds have passed.  Check non-required builds for any new compiler warnings.
* [ ] Sufficient testing is done. Check feature pragmas and test tags and that tests skipped for the ticket are run and now pass with the changes.
* [ ] If applicable, the PR has addressed any potential version compatibility issues.
* [ ] Check the target branch.   If it is master branch, should the PR go to a feature branch?  If it is a release branch, does it have merge approval in the JIRA ticket.
* [ ] Extra checks if forced landing is requested
  * [ ] Review comments are sufficiently resolved, particularly by prior reviewers that requested changes.
  * [ ] No new NLT or valgrind warnings.  Check the classic view.
  * [ ] Quick-build or Quick-functional is not used.
* [ ] Fix the commit message upon landing. Check the standard [here](https://daosio.atlassian.net/wiki/spaces/DC/pages/11133911069/Commit+Comments). Edit it to create a single commit. If necessary, ask submitter for a new summary.

dinghwah committed:
recommit

Skip-fnbullseye: false
Skip-bullseye: false
Skip-python-bandit: true
Skip-nlt: true
Skip-build-leap15-rpm: true
Skip-scan-leap15-rpms: true
Allow-unstable-test: true
Doc-only: false
Test-tag: sec_basic
Required-githooks: true
Signed-off-by: Ding Ho ding-hwa.ho@intel.com

dinghwah committed:
recommit

Skip-fnbullseye: false
Skip-bullseye: false
Skip-python-bandit: true
Skip-nlt: true
Skip-build-leap15-rpm: true
Skip-scan-leap15-rpms: true
Allow-unstable-test: true
Doc-only: false
Test-tag: sec_basic
Required-githooks: true
Signed-off-by: Ding Ho ding-hwa.ho@intel.com

dinghwah committed:
recommit

Skip-fnbullseye: false
Skip-bullseye: false
Skip-python-bandit: true
Skip-nlt: true
Skip-build-leap15-rpm: true
Skip-scan-leap15-rpms: true
Allow-unstable-test: true
Doc-only: false
Test-tag: sec_basic
Required-githooks: true
Signed-off-by: Ding Ho ding-hwa.ho@intel.com

dinghwah committed:
recommit

Skip-fnbullseye: false
Skip-bullseye: false
Skip-python-bandit: true
Skip-nlt: true
Skip-build-leap15-rpm: true
Skip-scan-leap15-rpms: true
Allow-unstable-test: true
Doc-only: false
Test-tag: sec_basic
Required-githooks: true
Signed-off-by: Ding Ho ding-hwa.ho@intel.com

dinghwah committed:
recommit

Skip-fnbullseye: false
Skip-bullseye: false
Skip-python-bandit: true
Skip-nlt: true
Skip-build-leap15-rpm: true
Skip-scan-leap15-rpms: true
Allow-unstable-test: true
Doc-only: false
Test-tag: sec_basic
Required-githooks: true
Signed-off-by: Ding Ho ding-hwa.ho@intel.com

dinghwah committed:
recommit

Skip-fnbullseye: false
Skip-bullseye: false
Skip-python-bandit: true
Skip-nlt: true
Skip-build-leap15-rpm: true
Skip-scan-leap15-rpms: true
Allow-unstable-test: true
Doc-only: false
Test-tag: sec_basic
Required-githooks: true
Signed-off-by: Ding Ho ding-hwa.ho@intel.com

dinghwah committed:
recommit

Skip-fnbullseye: false
Skip-bullseye: false
Skip-python-bandit: true
Skip-nlt: true
Skip-build-leap15-rpm: true
Skip-scan-leap15-rpms: true
Allow-unstable-test: true
Doc-only: false
Test-tag: sec_basic
Required-githooks: true
Signed-off-by: Ding Ho ding-hwa.ho@intel.com

dinghwah committed:
recommit

Skip-fnbullseye: false
Skip-bullseye: false
Skip-python-bandit: true
Skip-nlt: true
Skip-build-leap15-rpm: true
Skip-scan-leap15-rpms: true
Allow-unstable-test: true
Doc-only: false
Test-tag: sec_basic
Required-githooks: true
Signed-off-by: Ding Ho ding-hwa.ho@intel.com

dinghwah committed:
recommit

Skip-fnbullseye: false
Skip-bullseye: false
Skip-python-bandit: true
Skip-nlt: true
Skip-build-leap15-rpm: true
Skip-scan-leap15-rpms: true
Allow-unstable-test: true
Doc-only: false
Test-tag: sec_basic
Required-githooks: true
Signed-off-by: Ding Ho ding-hwa.ho@intel.com

dinghwah committed:
recommit

Skip-fnbullseye: false
Skip-bullseye: false
Skip-python-bandit: true
Skip-nlt: true
Skip-build-leap15-rpm: true
Skip-scan-leap15-rpms: true
Allow-unstable-test: true
Doc-only: false
Test-tag: sec_basic
Required-githooks: true
Signed-off-by: Ding Ho ding-hwa.ho@intel.com

dinghwah committed:
recommit

Skip-fnbullseye: false
Skip-bullseye: false
Skip-python-bandit: true
Skip-nlt: true
Skip-build-leap15-rpm: true
Skip-scan-leap15-rpms: true
Allow-unstable-test: true
Doc-only: false
Test-tag: sec_basic
Required-githooks: true
Signed-off-by: Ding Ho ding-hwa.ho@intel.com

dinghwah committed:
recommit

Skip-fnbullseye: false
Skip-bullseye: false
Skip-python-bandit: true
Skip-nlt: true
Skip-build-leap15-rpm: true
Skip-scan-leap15-rpms: true
Allow-unstable-test: true
Doc-only: false
Test-tag: sec_basic
Required-githooks: true
Signed-off-by: Ding Ho ding-hwa.ho@intel.com

dinghwah committed:
recommit

Skip-fnbullseye: false
Skip-bullseye: false
Skip-python-bandit: true
Skip-nlt: true
Skip-build-leap15-rpm: true
Skip-scan-leap15-rpms: true
Allow-unstable-test: true
Doc-only: false
Test-tag: sec_basic
Required-githooks: true
Signed-off-by: Ding Ho ding-hwa.ho@intel.com

dinghwah committed:
recommit

Skip-fnbullseye: false
Skip-bullseye: false
Skip-python-bandit: true
Skip-nlt: true
Skip-build-leap15-rpm: true
Skip-scan-leap15-rpms: true
Allow-unstable-test: true
Doc-only: false
Test-tag: sec_basic
Required-githooks: true
Signed-off-by: Ding Ho ding-hwa.ho@intel.com

dinghwah committed:
recommit

Skip-fnbullseye: false
Skip-bullseye: false
Skip-python-bandit: true
Skip-nlt: true
Skip-build-leap15-rpm: true
Skip-scan-leap15-rpms: true
Allow-unstable-test: true
Doc-only: false
Test-tag: sec_basic
Required-githooks: true
Signed-off-by: Ding Ho ding-hwa.ho@intel.com

dinghwah committed:
recommit

Skip-fnbullseye: false
Skip-bullseye: false
Skip-python-bandit: true
Skip-nlt: true
Skip-build-leap15-rpm: true
Skip-scan-leap15-rpms: true
Allow-unstable-test: true
Doc-only: false
Test-tag: sec_basic
Required-githooks: true
Signed-off-by: Ding Ho ding-hwa.ho@intel.com

dinghwah committed:
recommit

Skip-fnbullseye: false
Skip-bullseye: false
Skip-python-bandit: true
Skip-nlt: true
Skip-build-leap15-rpm: true
Skip-scan-leap15-rpms: true
Allow-unstable-test: true
Doc-only: false
Test-tag: sec_basic
Required-githooks: true
Signed-off-by: Ding Ho ding-hwa.ho@intel.com

dinghwah committed:
recommit

Skip-fnbullseye: false
Skip-bullseye: false
Skip-python-bandit: true
Skip-nlt: true
Skip-build-leap15-rpm: true
Skip-scan-leap15-rpms: true
Allow-unstable-test: true
Doc-only: false
Test-tag: sec_basic
Required-githooks: true
Signed-off-by: Ding Ho ding-hwa.ho@intel.com

dinghwah committed:
recommit

Skip-fnbullseye: false
Skip-bullseye: false
Skip-python-bandit: true
Skip-nlt: true
Skip-build-leap15-rpm: true
Skip-scan-leap15-rpms: true
Allow-unstable-test: true
Doc-only: false
Test-tag: sec_basic
Required-githooks: true
Signed-off-by: Ding Ho ding-hwa.ho@intel.com

dinghwah committed:
recommit

Skip-fnbullseye: false
Skip-bullseye: false
Skip-python-bandit: true
Skip-nlt: true
Skip-build-leap15-rpm: true
Skip-scan-leap15-rpms: true
Allow-unstable-test: true
Doc-only: false
Test-tag: sec_basic
Required-githooks: true
Signed-off-by: Ding Ho ding-hwa.ho@intel.com

dinghwah committed:
DAOS-15396 test: Coverage Report on Master March 2024

Test only, please do not merge.

Skip-fnbullseye: false
Skip-bullseye: false
Skip-python-bandit: true
Skip-build-EL9-rpm: true
Test-tag: pr daily_regression full_regression
Skip-func-hw-test-medium-md-on-ssd: false
Skip-func-hw-test-medium-verbs-provider-md-on-ssd: false
Skip-func-hw-test-medium-ucx-provider: false
Skip-func-hw-test-large-md-on-ssd: false
Allow-unstable-test: true
Doc-only: false
Required-githooks: true
Signed-off-by: Ding Ho ding-hwa.ho@intel.com

Skip-fnbullseye: false
Skip-bullseye: false
Skip-python-bandit: true
Skip-build-EL9-rpm: true
Allow-unstable-test: true
Doc-only: false
Test-nvme: auto_md_on_ssd
Test-tag: pr daily_regression full_regression
Required-githooks: true
Signed-off-by: Ding Ho ding-hwa.ho@intel.com

dinghwah committed:
DAOS-15396 test: Coverage Report on Master March 2024

Test only, please do not merge.

Skip-fnbullseye: false
Skip-bullseye: false
Skip-python-bandit: true
Skip-build-EL9-rpm: true
Test-tag: pr daily_regression full_regression
Skip-func-hw-test-medium-md-on-ssd: false
Skip-func-hw-test-medium-verbs-provider-md-on-ssd: false
Skip-func-hw-test-medium-ucx-provider: false
Skip-func-hw-test-large-md-on-ssd: false
Allow-unstable-test: true
Doc-only: false
Required-githooks: true
Signed-off-by: Ding Ho ding-hwa.ho@intel.com

Skip-fnbullseye: false
Skip-bullseye: false
Skip-python-bandit: true
Skip-build-EL9-rpm: true
Allow-unstable-test: true
Doc-only: false
Test-nvme: auto_md_on_ssd
Test-tag: pr daily_regression full_regression
Required-githooks: true
Signed-off-by: Ding Ho ding-hwa.ho@intel.com

dinghwah committed:
DAOS-15396 test: Coverage Report on Master March 2024

Test only, please do not merge.

Skip-fnbullseye: false
Skip-bullseye: false
Skip-python-bandit: true
Skip-build-EL9-rpm: true
Test-tag: pr daily_regression full_regression
Skip-func-hw-test-medium-md-on-ssd: false
Skip-func-hw-test-medium-verbs-provider-md-on-ssd: false
Skip-func-hw-test-medium-ucx-provider: false
Skip-func-hw-test-large-md-on-ssd: false
Allow-unstable-test: true
Doc-only: false
Required-githooks: true
Signed-off-by: Ding Ho ding-hwa.ho@intel.com

Skip-fnbullseye: false
Skip-bullseye: false
Skip-python-bandit: true
Skip-build-EL9-rpm: true
Allow-unstable-test: true
Doc-only: false
Test-nvme: auto_md_on_ssd
Test-tag: pr daily_regression full_regression
Required-githooks: true
Signed-off-by: Ding Ho ding-hwa.ho@intel.com

dinghwah committed:
DAOS-15396 test: Coverage Report on Master March 2024

Test only, please do not merge.

Skip-fnbullseye: false
Skip-bullseye: false
Skip-python-bandit: true
Skip-build-EL9-rpm: true
Test-tag: pr daily_regression full_regression
Skip-func-hw-test-medium-md-on-ssd: false
Skip-func-hw-test-medium-verbs-provider-md-on-ssd: false
Skip-func-hw-test-medium-ucx-provider: false
Skip-func-hw-test-large-md-on-ssd: false
Allow-unstable-test: true
Doc-only: false
Required-githooks: true
Signed-off-by: Ding Ho ding-hwa.ho@intel.com

Skip-fnbullseye: false
Skip-bullseye: false
Skip-python-bandit: true
Skip-build-EL9-rpm: true
Allow-unstable-test: true
Doc-only: false
Test-nvme: auto_md_on_ssd
Test-tag: pr daily_regression full_regression
Required-githooks: true
Signed-off-by: Ding Ho ding-hwa.ho@intel.com

dinghwah committed:
DAOS-15396 test: Coverage Report on Master March 2024

Test only, please do not merge.

Skip-fnbullseye: false
Skip-bullseye: false
Skip-python-bandit: true
Skip-build-EL9-rpm: true
Test-tag: pr daily_regression full_regression
Skip-func-hw-test-medium-md-on-ssd: false
Skip-func-hw-test-medium-verbs-provider-md-on-ssd: false
Skip-func-hw-test-medium-ucx-provider: false
Skip-func-hw-test-large-md-on-ssd: false
Allow-unstable-test: true
Doc-only: false
Required-githooks: true
Signed-off-by: Ding Ho ding-hwa.ho@intel.com

Skip-fnbullseye: false
Skip-bullseye: false
Skip-python-bandit: true
Skip-build-EL9-rpm: true
Allow-unstable-test: true
Doc-only: false
Test-nvme: auto_md_on_ssd
Test-tag: pr daily_regression full_regression
Required-githooks: true
Signed-off-by: Ding Ho ding-hwa.ho@intel.com

dinghwah committed:
DAOS-15396 test: Coverage Report on Master March 2024

Test only, please do not merge.

Skip-fnbullseye: false
Skip-bullseye: false
Skip-python-bandit: true
Skip-build-EL9-rpm: true
Test-tag: pr daily_regression full_regression
Skip-func-hw-test-medium-md-on-ssd: false
Skip-func-hw-test-medium-verbs-provider-md-on-ssd: false
Skip-func-hw-test-medium-ucx-provider: false
Skip-func-hw-test-large-md-on-ssd: false
Allow-unstable-test: true
Doc-only: false
Required-githooks: true
Signed-off-by: Ding Ho ding-hwa.ho@intel.com

Skip-fnbullseye: false
Skip-bullseye: false
Skip-python-bandit: true
Skip-build-EL9-rpm: true
Allow-unstable-test: true
Doc-only: false
Test-nvme: auto_md_on_ssd
Test-tag: pr daily_regression full_regression
Required-githooks: true
Signed-off-by: Ding Ho ding-hwa.ho@intel.com

dinghwah committed:
DAOS-15396 test: Coverage Report on Master March 2024

Test only, please do not merge.

Skip-fnbullseye: false
Skip-bullseye: false
Skip-python-bandit: true
Skip-build-EL9-rpm: true
Test-tag: pr daily_regression full_regression
Skip-func-hw-test-medium-md-on-ssd: false
Skip-func-hw-test-medium-verbs-provider-md-on-ssd: false
Skip-func-hw-test-medium-ucx-provider: false
Skip-func-hw-test-large-md-on-ssd: false
Allow-unstable-test: true
Doc-only: false
Required-githooks: true
Signed-off-by: Ding Ho ding-hwa.ho@intel.com

Skip-fnbullseye: false
Skip-bullseye: false
Skip-python-bandit: true
Skip-build-EL9-rpm: true
Allow-unstable-test: true
Doc-only: false
Test-nvme: auto_md_on_ssd
Test-tag: pr daily_regression full_regression
Required-githooks: true
Signed-off-by: Ding Ho ding-hwa.ho@intel.com

dinghwah committed:
DAOS-15396 test: Coverage Report on Master March 2024

Test only, please do not merge.

Skip-fnbullseye: false
Skip-bullseye: false
Skip-python-bandit: true
Skip-build-EL9-rpm: true
Test-tag: pr daily_regression full_regression
Skip-func-hw-test-medium-md-on-ssd: false
Skip-func-hw-test-medium-verbs-provider-md-on-ssd: false
Skip-func-hw-test-medium-ucx-provider: false
Skip-func-hw-test-large-md-on-ssd: false
Allow-unstable-test: true
Doc-only: false
Required-githooks: true
Signed-off-by: Ding Ho ding-hwa.ho@intel.com

Skip-fnbullseye: false
Skip-bullseye: false
Skip-python-bandit: true
Skip-build-EL9-rpm: true
Allow-unstable-test: true
Doc-only: false
Test-nvme: auto_md_on_ssd
Test-tag: pr daily_regression full_regression
Required-githooks: true
Signed-off-by: Ding Ho ding-hwa.ho@intel.com

dinghwah committed:
DAOS-15396 test: Coverage Report on Master March 2024

Test only, please do not merge.

Skip-fnbullseye: false
Skip-bullseye: false
Skip-python-bandit: true
Skip-build-EL9-rpm: true
Test-tag: pr daily_regression full_regression
Skip-func-hw-test-medium-md-on-ssd: false
Skip-func-hw-test-medium-verbs-provider-md-on-ssd: false
Skip-func-hw-test-medium-ucx-provider: false
Skip-func-hw-test-large-md-on-ssd: false
Allow-unstable-test: true
Doc-only: false
Required-githooks: true
Signed-off-by: Ding Ho ding-hwa.ho@intel.com

Skip-fnbullseye: false
Skip-bullseye: false
Skip-python-bandit: true
Skip-build-EL9-rpm: true
Allow-unstable-test: true
Doc-only: false
Test-nvme: auto_md_on_ssd
Test-tag: pr daily_regression full_regression
Required-githooks: true
Signed-off-by: Ding Ho ding-hwa.ho@intel.com

dinghwah committed:
DAOS-15396 test: Coverage Report on Master March 2024

Test only, please do not merge.

Skip-fnbullseye: false
Skip-bullseye: false
Skip-python-bandit: true
Skip-build-EL9-rpm: true
Test-tag: pr daily_regression full_regression
Skip-func-hw-test-medium-md-on-ssd: false
Skip-func-hw-test-medium-verbs-provider-md-on-ssd: false
Skip-func-hw-test-medium-ucx-provider: false
Skip-func-hw-test-large-md-on-ssd: false
Allow-unstable-test: true
Doc-only: false
Required-githooks: true
Signed-off-by: Ding Ho ding-hwa.ho@intel.com

Skip-fnbullseye: false
Skip-bullseye: false
Skip-python-bandit: true
Skip-build-EL9-rpm: true
Allow-unstable-test: true
Doc-only: false
Test-nvme: auto_md_on_ssd
Test-tag: pr daily_regression full_regression
Required-githooks: true
Signed-off-by: Ding Ho ding-hwa.ho@intel.com

dinghwah committed:
DAOS-15396 test: Coverage Report on Master March 2024

Test only, please do not merge.

Skip-fnbullseye: false
Skip-bullseye: false
Skip-python-bandit: true
Skip-build-EL9-rpm: true
Test-tag: pr daily_regression full_regression
Skip-func-hw-test-medium-md-on-ssd: false
Skip-func-hw-test-medium-verbs-provider-md-on-ssd: false
Skip-func-hw-test-medium-ucx-provider: false
Skip-func-hw-test-large-md-on-ssd: false
Allow-unstable-test: true
Doc-only: false
Required-githooks: true
Signed-off-by: Ding Ho ding-hwa.ho@intel.com

Skip-fnbullseye: false
Skip-bullseye: false
Skip-python-bandit: true
Skip-build-EL9-rpm: true
Allow-unstable-test: true
Doc-only: false
Test-nvme: auto_md_on_ssd
Test-tag: pr daily_regression full_regression
Required-githooks: true
Signed-off-by: Ding Ho ding-hwa.ho@intel.com

dinghwah committed:
DAOS-15396 test: Coverage Report on Master March 2024

Test only, please do not merge.

Skip-fnbullseye: false
Skip-bullseye: false
Skip-python-bandit: true
Skip-build-EL9-rpm: true
Test-tag: pr daily_regression full_regression
Skip-func-hw-test-medium-md-on-ssd: false
Skip-func-hw-test-medium-verbs-provider-md-on-ssd: false
Skip-func-hw-test-medium-ucx-provider: false
Skip-func-hw-test-large-md-on-ssd: false
Allow-unstable-test: true
Doc-only: false
Required-githooks: true
Signed-off-by: Ding Ho ding-hwa.ho@intel.com

Skip-fnbullseye: false
Skip-bullseye: false
Skip-python-bandit: true
Skip-build-EL9-rpm: true
Allow-unstable-test: true
Doc-only: false
Test-nvme: auto_md_on_ssd
Test-tag: pr daily_regression full_regression
Required-githooks: true
Signed-off-by: Ding Ho ding-hwa.ho@intel.com

dinghwah committed:
DAOS-15396 test: Coverage Report on Master March 2024

Test only, please do not merge.

Skip-fnbullseye: false
Skip-bullseye: false
Skip-python-bandit: true
Skip-build-EL9-rpm: true
Test-tag: pr daily_regression full_regression
Skip-func-hw-test-medium-md-on-ssd: false
Skip-func-hw-test-medium-verbs-provider-md-on-ssd: false
Skip-func-hw-test-medium-ucx-provider: false
Skip-func-hw-test-large-md-on-ssd: false
Allow-unstable-test: true
Doc-only: false
Required-githooks: true
Signed-off-by: Ding Ho ding-hwa.ho@intel.com

Skip-fnbullseye: false
Skip-bullseye: false
Skip-python-bandit: true
Skip-build-EL9-rpm: true
Allow-unstable-test: true
Doc-only: false
Test-nvme: auto_md_on_ssd
Test-tag: pr daily_regression full_regression
Required-githooks: true
Signed-off-by: Ding Ho ding-hwa.ho@intel.com

dinghwah committed:
DAOS-15396 test: Coverage Report on Master March 2024

Test only, please do not merge.

Skip-fnbullseye: false
Skip-bullseye: false
Skip-python-bandit: true
Skip-build-EL9-rpm: true
Test-tag: pr daily_regression full_regression
Skip-func-hw-test-medium-md-on-ssd: false
Skip-func-hw-test-medium-verbs-provider-md-on-ssd: false
Skip-func-hw-test-medium-ucx-provider: false
Skip-func-hw-test-large-md-on-ssd: false
Allow-unstable-test: true
Doc-only: false
Required-githooks: true
Signed-off-by: Ding Ho ding-hwa.ho@intel.com

Skip-fnbullseye: false
Skip-bullseye: false
Skip-python-bandit: true
Skip-build-EL9-rpm: true
Allow-unstable-test: true
Doc-only: false
Test-nvme: auto_md_on_ssd
Test-tag: pr daily_regression full_regression
Required-githooks: true
Signed-off-by: Ding Ho ding-hwa.ho@intel.com

dinghwah committed:
DAOS-15396 test: Coverage Report on Master March 2024

Test only, please do not merge.

Skip-fnbullseye: false
Skip-bullseye: false
Skip-python-bandit: true
Skip-build-EL9-rpm: true
Test-tag: pr daily_regression full_regression
Skip-func-hw-test-medium-md-on-ssd: false
Skip-func-hw-test-medium-verbs-provider-md-on-ssd: false
Skip-func-hw-test-medium-ucx-provider: false
Skip-func-hw-test-large-md-on-ssd: false
Allow-unstable-test: true
Doc-only: false
Required-githooks: true
Signed-off-by: Ding Ho ding-hwa.ho@intel.com

Skip-fnbullseye: false
Skip-bullseye: false
Skip-python-bandit: true
Skip-build-EL9-rpm: true
Allow-unstable-test: true
Doc-only: false
Test-nvme: auto_md_on_ssd
Test-tag: pr daily_regression full_regression
Required-githooks: true
Signed-off-by: Ding Ho ding-hwa.ho@intel.com

dinghwah committed:
DAOS-15396 test: Coverage Report on Master March 2024

Test only, please do not merge.

Skip-fnbullseye: false
Skip-bullseye: false
Skip-python-bandit: true
Skip-build-EL9-rpm: true
Test-tag: pr daily_regression full_regression
Skip-func-hw-test-medium-md-on-ssd: false
Skip-func-hw-test-medium-verbs-provider-md-on-ssd: false
Skip-func-hw-test-medium-ucx-provider: false
Skip-func-hw-test-large-md-on-ssd: false
Allow-unstable-test: true
Doc-only: false
Required-githooks: true
Signed-off-by: Ding Ho ding-hwa.ho@intel.com

Skip-fnbullseye: false
Skip-bullseye: false
Skip-python-bandit: true
Skip-build-EL9-rpm: true
Allow-unstable-test: true
Doc-only: false
Test-nvme: auto_md_on_ssd
Test-tag: pr daily_regression full_regression
Required-githooks: true
Signed-off-by: Ding Ho ding-hwa.ho@intel.com

dinghwah committed:
DAOS-15396 test: Coverage Report on Master March 2024

Test only, please do not merge.

Skip-fnbullseye: false
Skip-bullseye: false
Skip-python-bandit: true
Skip-build-EL9-rpm: true
Test-tag: pr daily_regression full_regression
Skip-func-hw-test-medium-md-on-ssd: false
Skip-func-hw-test-medium-verbs-provider-md-on-ssd: false
Skip-func-hw-test-medium-ucx-provider: false
Skip-func-hw-test-large-md-on-ssd: false
Allow-unstable-test: true
Doc-only: false
Required-githooks: true
Signed-off-by: Ding Ho ding-hwa.ho@intel.com

Skip-fnbullseye: false
Skip-bullseye: false
Skip-python-bandit: true
Skip-build-EL9-rpm: true
Allow-unstable-test: true
Doc-only: false
Test-nvme: auto_md_on_ssd
Test-tag: pr daily_regression full_regression
Required-githooks: true
Signed-off-by: Ding Ho ding-hwa.ho@intel.com

dinghwah committed:
DAOS-15396 test: Coverage Report on Master March 2024

Test only, please do not merge.

Skip-fnbullseye: false
Skip-bullseye: false
Skip-python-bandit: true
Skip-build-EL9-rpm: true
Test-tag: pr daily_regression full_regression
Skip-func-hw-test-medium-md-on-ssd: false
Skip-func-hw-test-medium-verbs-provider-md-on-ssd: false
Skip-func-hw-test-medium-ucx-provider: false
Skip-func-hw-test-large-md-on-ssd: false
Allow-unstable-test: true
Doc-only: false
Required-githooks: true
Signed-off-by: Ding Ho ding-hwa.ho@intel.com

Skip-fnbullseye: false
Skip-bullseye: false
Skip-python-bandit: true
Skip-build-EL9-rpm: true
Allow-unstable-test: true
Doc-only: false
Test-nvme: auto_md_on_ssd
Test-tag: pr daily_regression full_regression
Required-githooks: true
Signed-off-by: Ding Ho ding-hwa.ho@intel.com

dinghwah committed:
DAOS-15396 test: Coverage Report on Master March 2024

Test only, please do not merge.

Skip-fnbullseye: false
Skip-bullseye: false
Skip-python-bandit: true
Skip-build-EL9-rpm: true
Test-tag: pr daily_regression full_regression
Skip-func-hw-test-medium-md-on-ssd: false
Skip-func-hw-test-medium-verbs-provider-md-on-ssd: false
Skip-func-hw-test-medium-ucx-provider: false
Skip-func-hw-test-large-md-on-ssd: false
Allow-unstable-test: true
Doc-only: false
Required-githooks: true
Signed-off-by: Ding Ho ding-hwa.ho@intel.com

Skip-fnbullseye: false
Skip-bullseye: false
Skip-python-bandit: true
Skip-build-EL9-rpm: true
Allow-unstable-test: true
Doc-only: false
Test-nvme: auto_md_on_ssd
Test-tag: pr daily_regression full_regression
Required-githooks: true
Signed-off-by: Ding Ho ding-hwa.ho@intel.com

dinghwah committed:
DAOS-15396 test: Coverage Report on Master March 2024

Test only, please do not merge.

Skip-fnbullseye: false
Skip-bullseye: false
Skip-python-bandit: true
Skip-build-EL9-rpm: true
Test-tag: pr daily_regression full_regression
Skip-func-hw-test-medium-md-on-ssd: false
Skip-func-hw-test-medium-verbs-provider-md-on-ssd: false
Skip-func-hw-test-medium-ucx-provider: false
Skip-func-hw-test-large-md-on-ssd: false
Allow-unstable-test: true
Doc-only: false
Required-githooks: true
Signed-off-by: Ding Ho ding-hwa.ho@intel.com

Skip-fnbullseye: false
Skip-bullseye: false
Skip-python-bandit: true
Skip-build-EL9-rpm: true
Allow-unstable-test: true
Doc-only: false
Test-nvme: auto_md_on_ssd
Test-tag: pr daily_regression full_regression
Required-githooks: true
Signed-off-by: Ding Ho ding-hwa.ho@intel.com

dinghwah committed:
retore pool_security_test_base.py

Required-githooks: true

dinghwah committed:
retore pool_security_test_base.py

Required-githooks: true

dinghwah committed:
retore pool_security_test_base.py

Required-githooks: true

dinghwah committed:
retore pool_security_test_base.py

Required-githooks: true

dinghwah committed:
retore pool_security_test_base.py

Required-githooks: true

dinghwah committed:
retore pool_security_test_base.py

Required-githooks: true

dinghwah committed:
retore pool_security_test_base.py

Required-githooks: true

dinghwah committed:
retore pool_security_test_base.py

Required-githooks: true

dinghwah committed:
retore pool_security_test_base.py

Required-githooks: true

dinghwah committed:
retore pool_security_test_base.py

Required-githooks: true

dinghwah committed:
retore pool_security_test_base.py

Required-githooks: true

dinghwah committed:
retore pool_security_test_base.py

Required-githooks: true

dinghwah committed:
retore pool_security_test_base.py

Required-githooks: true

dinghwah committed:
retore pool_security_test_base.py

Required-githooks: true

dinghwah committed:
retore pool_security_test_base.py

Required-githooks: true

dinghwah committed:
retore pool_security_test_base.py

Required-githooks: true

dinghwah committed:
retore pool_security_test_base.py

Required-githooks: true

dinghwah committed:
retore pool_security_test_base.py

Required-githooks: true

dinghwah committed:
retore pool_security_test_base.py

Required-githooks: true

dinghwah committed:
retore pool_security_test_base.py

Required-githooks: true

dinghwah committed:
restore pool_security_test_base.py

Required-githooks: true

dinghwah committed:
restore pool_security_test_base.py

Required-githooks: true

dinghwah committed:
restore pool_security_test_base.py

Required-githooks: true

dinghwah committed:
restore pool_security_test_base.py

Required-githooks: true

dinghwah committed:
restore pool_security_test_base.py

Required-githooks: true

dinghwah committed:
restore pool_security_test_base.py

Required-githooks: true

dinghwah committed:
restore pool_security_test_base.py

Required-githooks: true

dinghwah committed:
restore pool_security_test_base.py

Required-githooks: true

dinghwah committed:
restore pool_security_test_base.py

Required-githooks: true

dinghwah committed:
restore pool_security_test_base.py

Required-githooks: true

dinghwah committed:
restore pool_security_test_base.py

Required-githooks: true

dinghwah committed:
restore pool_security_test_base.py

Required-githooks: true

dinghwah committed:
restore pool_security_test_base.py

Required-githooks: true

dinghwah committed:
restore pool_security_test_base.py

Required-githooks: true

dinghwah committed:
restore pool_security_test_base.py

Required-githooks: true

dinghwah committed:
restore pool_security_test_base.py

Required-githooks: true

dinghwah committed:
restore pool_security_test_base.py

Required-githooks: true

dinghwah committed:
restore pool_security_test_base.py

Required-githooks: true

dinghwah committed:
restore pool_security_test_base.py

Required-githooks: true

dinghwah committed:
restore pool_security_test_base.py

Required-githooks: true

dinghwah committed:
Restore pool_security_test_base.py

Required-githooks: true

dinghwah committed:
Restore pool_security_test_base.py

Required-githooks: true

dinghwah committed:
Restore pool_security_test_base.py

Required-githooks: true

dinghwah committed:
Restore pool_security_test_base.py

Required-githooks: true

dinghwah committed:
Restore pool_security_test_base.py

Required-githooks: true

dinghwah committed:
Restore pool_security_test_base.py

Required-githooks: true

dinghwah committed:
Restore pool_security_test_base.py

Required-githooks: true

dinghwah committed:
Restore pool_security_test_base.py

Required-githooks: true

dinghwah committed:
Restore pool_security_test_base.py

Required-githooks: true

dinghwah committed:
Restore pool_security_test_base.py

Required-githooks: true

dinghwah committed:
DAOS-15937 test: Automate metadata duplicate rpc detection time consuming

Skip-unit-tests: true
Skip-fault-injection-test: true
Skip-func-hw-test-large-md-on-ssd: false
Test-tag: test_metadata_dup_rpc
Doc-only: false
Required-githooks: true

Signed-off-by: Ding Ho <ding-hwa.ho@intel.com>

dinghwah committed:
Update metadata_svc_ops.py

Skip-unit-tests: true
Skip-fault-injection-test: true
Skip-func-hw-test-large-md-on-ssd: false
Test-tag: test_metadata_dup_rpc
Doc-only: false
Required-githooks: true
Signed-off-by: Ding Ho <ding-hwa.ho@intel.com>

daosbuild1 commented:
Test stage Functional Hardware Medium completed with status UNSTABLE.  https://build.hpdd.intel.com/job/daos-stack/job/daos//view/change-requests/job/PR-14473/2/testReport/

dinghwah committed:
Update metadata_svc_ops.yaml with manual tests.

Skip-unit-tests: true
Skip-fault-injection-test: true
Skip-func-hw-test-medium-md-on-ssd: false
Test-tag: test_metadata_dup_rpc
Doc-only: false
Required-githooks: true
Signed-off-by: Ding Ho <ding-hwa.ho@intel.com>

kccain commented on src/tests/ftest/server/metadata_svc_ops.py:
```diff
@@ -0,0 +1,139 @@
+        self.log.info("pool1 results = %s", results[0].result)
+        self.log.info("baseline results = %s", base_results[0].result)
+        self.log.info("average baseline result= %s", average_time)
+        for result in results[0].result:
```
Test steps documentation for step 4 should not specify that it is calculating an average for those loops following svc_ops_entry_age time.

This verification seems OK, since we expect the "early" loops before svc_ops_entry_age to be quicker than what the performance will "stabilize" to after svc_ops_entry_age time. i.e., all timings before and after svc_ops_entry_age time should fit underneath the baseline times the threshold_factor. But we particularly care about the iterations after svc_ops_entry_age time has passed.

dinghwah replied:
Please see the new commit, which should have addressed all the comments.

kccain reviewed (commented):
first pass review

dinghwah committed:
Update test script and yaml per Ken's comments.

Skip-unit-tests: true
Skip-fault-injection-test: true
Skip-func-hw-test-medium-md-on-ssd: false
Test-tag: test_metadata_dup_rpc
Doc-only: false
Required-githooks: true
Signed-off-by: Ding Ho <ding-hwa.ho@intel.com>

dinghwah committed:
Merge branch 'master' into dinghwah/DAOS-15937-mdtest

Skip-unit-tests: true
Skip-fault-injection-test: true
Skip-build-ubuntu20-rpm: true
Skip-func-hw-test-medium-md-on-ssd: false
Test-tag: test_metadata_dup_rpc
Doc-only: false
Required-githooks: true
Signed-off-by: Ding Ho <ding-hwa.ho@intel.com>

dinghwah committed:
Merge branch 'master' into dinghwah/DAOS-15937-mdtest

Skip-unit-tests: true
Skip-fault-injection-test: true
Skip-func-hw-test-medium-md-on-ssd: false
Test-tag: test_metadata_dup_rpc
Doc-only: false
Required-githooks: true
Signed-off-by: Ding Ho <ding-hwa.ho@intel.com>

dinghwah commented on src/tests/ftest/server/metadata_svc_ops.yaml:
```diff
@@ -0,0 +1,31 @@
+timeout: 850
+server_config:
+  name: daos_server
+  engines_per_host: 1
```
Updated with 2 engines per host.

dinghwah commented on src/tests/ftest/server/metadata_svc_ops.py:
```diff
@@ -0,0 +1,139 @@
+        :avocado: tags=server,metadata
+        :avocado: tags=DuplicateRpcDetection,test_metadata_dup_rpc
+        """
+        self.dmg.server_set_logmasks("DEBUG", raise_exception=False)
```
Updated.  server log mask defined by test yaml.

dinghwah commented on src/tests/ftest/server/metadata_svc_ops.py:
```diff
@@ -0,0 +1,139 @@
+        threshold_factor = self.params.get("threshold_factor", '/run/metadata/*', default=1.75)
+
+        self.log_step("Create pool with properties svc_ops_entry_age.")
+        pool1 = self.get_pool(dmg=self.dmg.copy())
```
yes. pool property svc_ops_entry_age handled by test yaml.

dinghwah commented on src/tests/ftest/server/metadata_svc_ops.py:
```diff
@@ -0,0 +1,139 @@
+        self.log_step("Create containers by ThreadManager.")
+        container_manager = ThreadManager(
+            self.metadata_workload_test, self.get_remaining_time() - 30)
+        container_manager.add(
```
Thanks for the comment.
When multiple threads specified (in testcase-2, 2, 4, or 8 threads), each thread will create it's own container and perform their individual metadata workload. 
Have updated.  It should be number_of_container (instead of cont_num).

kccain replied:
If the yaml file specifies number_thread: 2 (anything larger than 1), then this code will need to change so that it has a loop to call container_manager.add() calls so that all of the threads get created, is that right?

daltonbohning replied:
I agree with @kccain here. If `num_of_cont` is increased, this still only runs a single thread because the code only calls `container_manager.add` one time

> script will generate different number of container-thread (2, 4 or 8) on the same pool (thread)

Where? What line(s) of code does this?

dinghwah replied:
Line #107 to #110 generate different number of container-thread (2, 4 or 8) on the same pool.

daltonbohning replied:
```
        container_manager = ThreadManager(
            self.metadata_workload_test, self.get_remaining_time() - 30)
        container_manager.add(
            pool=self.pool, num_of_cont=num_of_cont, workload_cycles=w_cycles, test_loops=t_loops)
```


That is one thread, no matter what `num_of_cont` is

kccain commented on src/tests/ftest/server/metadata_svc_ops.py:
```diff
@@ -0,0 +1,143 @@
+            1. Bring up DAOS server.
+            2. Create pool1 with specified property svc_ops_entry_age.
+            3. Create containers by ThreadManager.
+            4. Run specified metadata workload cycles in multiple test loops (N cycles per loop),
```
```suggestion
            4. Run specified metadata workload cycles in multiple test loops (N cycles per loop).
```

kccain replied:
and suggest removing the next 2 lines of text, because the average time is not calculated.

dinghwah replied:
Thanks for the comment. 
Please see the new commit.

dinghwah committed:
Update metadata_svc_ops.py per Ken's comments.

Skip-unit-tests: true
Skip-fault-injection-test: true
Skip-func-hw-test-medium-md-on-ssd: false
Test-tag: test_metadata_dup_rpc
Doc-only: false
Required-githooks: true
Signed-off-by: Ding Ho <ding-hwa.ho@intel.com>

daosbuild1 commented:
Test stage Python Bandit check completed with status FAILURE.  https://build.hpdd.intel.com//job/daos-stack/job/daos/view/change-requests/job/PR-14473/7/execution/node/143/log

daosbuild1 commented:
Test stage Build RPM on EL 8 completed with status FAILURE.  https://build.hpdd.intel.com//job/daos-stack/job/daos/view/change-requests/job/PR-14473/7/execution/node/267/log

daosbuild1 commented:
Test stage Build RPM on EL 9 completed with status FAILURE.  https://build.hpdd.intel.com//job/daos-stack/job/daos/view/change-requests/job/PR-14473/7/execution/node/328/log

daosbuild1 commented:
Test stage Build RPM on Leap 15.5 completed with status FAILURE.  https://build.hpdd.intel.com//job/daos-stack/job/daos/view/change-requests/job/PR-14473/7/execution/node/329/log

dinghwah committed:
Merge branch 'master' into dinghwah/DAOS-15937-mdtest

Skip-unit-tests: true
Skip-fault-injection-test: true
Skip-func-hw-test-medium-md-on-ssd: false
Test-tag: test_metadata_dup_rpc
Doc-only: false
Required-githooks: true
Signed-off-by: Ding Ho <ding-hwa.ho@intel.com>

daosbuild1 commented:
Test stage Python Bandit check completed with status FAILURE.  https://build.hpdd.intel.com//job/daos-stack/job/daos/view/change-requests/job/PR-14473/8/execution/node/143/log

daosbuild1 commented:
Test stage Build RPM on EL 8 completed with status FAILURE.  https://build.hpdd.intel.com//job/daos-stack/job/daos/view/change-requests/job/PR-14473/8/execution/node/317/log

daosbuild1 commented:
Test stage Build RPM on EL 9 completed with status FAILURE.  https://build.hpdd.intel.com//job/daos-stack/job/daos/view/change-requests/job/PR-14473/8/execution/node/381/log

daosbuild1 commented:
Test stage Build RPM on Leap 15.5 completed with status FAILURE.  https://build.hpdd.intel.com//job/daos-stack/job/daos/view/change-requests/job/PR-14473/8/execution/node/353/log

daosbuild1 commented:
Test stage Build DEB on Ubuntu 20.04 completed with status FAILURE.  https://build.hpdd.intel.com//job/daos-stack/job/daos/view/change-requests/job/PR-14473/8/execution/node/285/log

dinghwah committed:
Merge branch 'master' into dinghwah/DAOS-15937-mdtest

Skip-unit-tests: true
Skip-fault-injection-test: true
Skip-func-hw-test-medium-md-on-ssd: false
Test-tag: test_metadata_dup_rpc
Doc-only: false
Required-githooks: true
Signed-off-by: Ding Ho <ding-hwa.ho@intel.com>

daltonbohning commented on src/tests/ftest/server/metadata_svc_ops.py:
```diff
@@ -0,0 +1,141 @@
+        threshold_factor = self.params.get("threshold_factor", '/run/metadata/*', default=1.75)
+
+        self.log_step("Create pool with properties svc_ops_entry_age.")
+        pool1 = self.get_pool(dmg=self.dmg.copy())
```
We can just get the command here
```suggestion
        pool1 = self.get_pool(dmg=self.get_dmg_command().copy())
```

daltonbohning commented on src/tests/ftest/server/metadata_svc_ops.py:
```diff
@@ -0,0 +1,141 @@
+        """
+        test_time = []
+        try:
+            container = self.get_container(pool, create=False)
+            if container.daos:
+                container.daos.verbose = False
+            container.create()
```
It's better to do this upfront

```suggestion
            daos = self.get_daos_command()
            daos.verbose = False
            container = self.get_container(pool, daos=daos)
```

daltonbohning commented on src/tests/ftest/server/metadata_svc_ops.py:
```diff
@@ -0,0 +1,141 @@
+        for ind in range(test_loops):
+            start = time.time()
+            for _ in range(workload_cycles):
+                container.close()
+                container.open()
```
Shouldn't this instead `open` and then `close`? The container normally wouldn't be open to begin with

daltonbohning commented on src/tests/ftest/server/metadata_svc_ops.py:
```diff
@@ -0,0 +1,141 @@
+        for ind in range(test_loops):
+            self.log.info("Test time of Metadata test-loop: %d,  %f",
+                          ind, test_time[ind])
+        container.close()
```
And then we shouldn't need this

daltonbohning commented on src/tests/ftest/server/metadata_svc_ops.py:
```diff
@@ -0,0 +1,141 @@
+            "To establish a baseline time without duplicate rpc detection, ",
+            "calculating average time per loop executed on pool2.")
+        base_results = container_manager.run()
+        base_average_time = statistics.mean(base_results[0].result)
+        num_failed = len(list(filter(lambda r: not r.passed, base_results)))
+        if num_failed > 0:
+            self.fail('#{} container create threads failed'.format(num_failed))
```
Should we check for failure before getting the mean? In case the mean can't be calculated from some bad result?
```suggestion
        num_failed = len(list(filter(lambda r: not r.passed, base_results)))
        if num_failed > 0:
            self.fail('#{} container create threads failed'.format(num_failed))
        base_average_time = statistics.mean(base_results[0].result)
```

dinghwah committed:
Update script per Dalton's comments.

Skip-unit-tests: true
Skip-fault-injection-test: true
Skip-func-hw-test-medium-md-on-ssd: false
Test-tag: test_metadata_dup_rpc
Doc-only: false
Required-githooks: true
Signed-off-by: Ding Ho <ding-hwa.ho@intel.com>

dinghwah commented:
Please review the new updates.

dinghwah committed:
Merge branch 'master' into dinghwah/DAOS-15937-mdtest
Required-githooks: true

dinghwah committed:
Update metadata_svc_ops.py per Dalton's comment.

Skip-unit-tests: true
Skip-fault-injection-test: true
Skip-func-hw-test-medium-md-on-ssd: false
Test-tag: test_metadata_dup_rpc
Doc-only: false
Required-githooks: true
Signed-off-by: Ding Ho <ding-hwa.ho@intel.com>

dinghwah committed:
Merge branch 'master' into dinghwah/DAOS-15937-mdtest

Skip-unit-tests: true
Skip-fault-injection-test: true
Skip-func-hw-test-medium-md-on-ssd: false
Test-tag: test_metadata_dup_rpc
Doc-only: false
Required-githooks: true
Signed-off-by: Ding Ho <ding-hwa.ho@intel.com>

dinghwah committed:
Update metadata_svc_ops.py for pylint.

Skip-unit-tests: true
Skip-fault-injection-test: true
Skip-func-hw-test-medium-md-on-ssd: false
Test-tag: test_metadata_dup_rpc
Doc-only: false
Required-githooks: true
Signed-off-by: Ding Ho <ding-hwa.ho@intel.com>

daosbuild1 commented:
Test stage Functional Hardware Medium completed with status UNSTABLE.  https://build.hpdd.intel.com/job/daos-stack/job/daos//view/change-requests/job/PR-14473/12/testReport/

dinghwah committed:
Recommit with factor to 2.5

Skip-unit-tests: true
Skip-fault-injection-test: true
Skip-func-hw-test-medium-md-on-ssd: false
Test-tag: test_metadata_dup_rpc
Test-repeat: 5
Doc-only: false
Required-githooks: true
Signed-off-by: Ding Ho <ding-hwa.ho@intel.com>

dinghwah committed:
Merge branch 'master' into dinghwah/DAOS-15937-mdtest

Skip-unit-tests: true
Skip-fault-injection-test: true
Skip-func-hw-test-medium-md-on-ssd: false
Test-tag: test_metadata_dup_rpc
Test_repeat: 5
Doc-only: false
Required-githooks: true
Signed-off-by: Ding Ho <ding-hwa.ho@intel.com>

kccain commented on src/tests/ftest/server/metadata_svc_ops.yaml:
```diff
@@ -0,0 +1,56 @@
+      fabric_iface: ib1
+      fabric_iface_port: 31417
+      log_file: daos_server1.log
+      log_mask: DEBUG,MEM=ERR
```
want to have log_mask: ERR when running a performance-sensitive test

dinghwah replied:
Please see the new commit.

dinghwah commented on src/tests/ftest/server/metadata_svc_ops.yaml:
```diff
@@ -0,0 +1,56 @@
+      fabric_iface: ib0
+      fabric_iface_port: 31317
+      log_file: daos_server0.log
+      log_mask: DEBUG,MEM=ERR
```
Please see the new commit.

dinghwah committed:
Merge branch 'master' into dinghwah/DAOS-15937-mdtest

dinghwah committed:
Update metadata_svc_ops.yaml per Ken's comment.

Skip-unit-tests: true
Skip-fault-injection-test: true
Skip-func-hw-test-medium-md-on-ssd: false
Test-tag: test_metadata_dup_rpc
Test-repeat: 10
Doc-only: false
Required-githooks: true
Signed-off-by: Ding Ho <ding-hwa.ho@intel.com>

Signed-off-by: Ding Ho <ding-hwa.ho@intel.com>

daosbuild1 commented:
Test stage Functional Hardware Medium MD on SSD completed with status UNSTABLE.  https://build.hpdd.intel.com/job/daos-stack/job/daos//view/change-requests/job/PR-14473/14/testReport/

daosbuild1 commented:
Test stage Functional Hardware Medium completed with status UNSTABLE.  https://build.hpdd.intel.com/job/daos-stack/job/daos//view/change-requests/job/PR-14473/14/testReport/

kccain commented:
> Test stage Functional Hardware Medium MD on SSD completed with status UNSTABLE. https://build.hpdd.intel.com/job/daos-stack/job/daos//view/change-requests/job/PR-14473/14/testReport/

The job.log has 200,000 lines of output, 100,000 pairs of items like this, that are output during the timing loops. We may want to look at making the output quiet. I think I had experimented with this at some point, but not sure if I can find the prototype code I used since it has been a while.
```
2024-08-23 08:32:00,394 test_utils_conta L0624 INFO | Opening container 138475AD-324F-4A86-8E25-D2D213FAABEB
2024-08-23 08:32:00,399 test_utils_conta L0645 INFO | Closing container 138475AD-324F-4A86-8E25-D2D213FAABEB

```

Failure reason
`#Dup rpc detection time 13.781859397888184 > baseline_time 5.202355599403381 * threshold_factor 2.5
`
but the actual time comparison should be 5.x versus 19..x seconds, which is the "stabilized" time after getting through the initial svc_ops_entry_age:60 seconds period

From the[ job.log](https://build.hpdd.intel.com/job/daos-stack/job/daos/view/change-requests/job/PR-14473/14/artifact/Functional%20Hardware%20Medium%20MD%20on%20SSD/server/metadata_svc_ops.py/repeat001/job.log), the first 10 test-loops with svc_ops_enabled:1
```
2024-08-23 08:34:54,578 metadata_svc_ops L0060 INFO | Test time of Metadata test-loop: 0,  8.415847
2024-08-23 08:34:54,578 metadata_svc_ops L0060 INFO | Test time of Metadata test-loop: 1,  13.758781
2024-08-23 08:34:54,578 metadata_svc_ops L0060 INFO | Test time of Metadata test-loop: 2,  18.333476
2024-08-23 08:34:54,578 metadata_svc_ops L0060 INFO | Test time of Metadata test-loop: 3,  19.055028
2024-08-23 08:34:54,578 metadata_svc_ops L0060 INFO | Test time of Metadata test-loop: 4,  19.116618
2024-08-23 08:34:54,578 metadata_svc_ops L0060 INFO | Test time of Metadata test-loop: 5,  19.093971
2024-08-23 08:34:54,578 metadata_svc_ops L0060 INFO | Test time of Metadata test-loop: 6,  19.083379
2024-08-23 08:34:54,578 metadata_svc_ops L0060 INFO | Test time of Metadata test-loop: 7,  19.085228
2024-08-23 08:34:54,578 metadata_svc_ops L0060 INFO | Test time of Metadata test-loop: 8,  19.120708
2024-08-23 08:34:54,578 metadata_svc_ops L0060 INFO | Test time of Metadata test-loop: 9,  19.120152
```

dinghwah committed:
Update metadata_svc_ops.yaml with silent: true

Required-githooks: true

dinghwah committed:
Merge branch 'master' into dinghwah/DAOS-15937-mdtest

Skip-unit-tests: true
Skip-fault-injection-test: true
Skip-func-hw-test-medium-md-on-ssd: false
Test-tag: test_metadata_dup_rpc
Test-repeat: 10
Doc-only: false
Required-githooks: true
Signed-off-by: Ding Ho <ding-hwa.ho@intel.com>

daosbuild1 commented:
Test stage Functional Hardware Medium MD on SSD completed with status UNSTABLE.  https://build.hpdd.intel.com/job/daos-stack/job/daos//view/change-requests/job/PR-14473/15/testReport/

daosbuild1 commented:
Test stage Functional Hardware Medium completed with status UNSTABLE.  https://build.hpdd.intel.com/job/daos-stack/job/daos//view/change-requests/job/PR-14473/15/testReport/

dinghwah committed:
Update metadata_svc_ops.yaml to test svc-ops 150

Skip-unit-tests: true
Skip-fault-injection-test: true
Skip-func-hw-test-medium-md-on-ssd: false
Test-tag: test_metadata_dup_rpc
Test-repeat: 2
Doc-only: false
Required-githooks: true
Signed-off-by: Ding Ho <ding-hwa.ho@intel.com>

daosbuild1 commented:
Test stage Functional Hardware Medium MD on SSD completed with status UNSTABLE.  https://build.hpdd.intel.com/job/daos-stack/job/daos//view/change-requests/job/PR-14473/16/testReport/

daosbuild1 commented:
Test stage Functional Hardware Medium completed with status UNSTABLE.  https://build.hpdd.intel.com/job/daos-stack/job/daos//view/change-requests/job/PR-14473/16/testReport/

kccain commented:
updated my comment from a few days ago, after updating the debug PR to get a proper 1000 container open/close loop timing in the daos_test -c co_op_dup_timing() with svc_ops_enabled:1 and svc_ops_entry_age:60, and preceding the 1000 loop timing with a 60 second warmup loop.

The timings measured were 2.969 seconds (Functional HW Medium) and 1.617 seconds (Functional HW Medium MD on SSD)
